tests/test_custom_checkers/test_infinite_loop_checker.py

```diff
@@ -178,24 +176,6 @@ def test_while_func_call(self) -> None:
             self.checker.visit_while(node1)
             self.checker.visit_while(node2)
 
-    def test_while_fund_call_var(self) -> None:
```

Comment:
Hmmm, the fact that you deleted this test is a sign something weird happened.

Right now the logic is considering inference failures (returning `Uninferable`, `None`) as preventing the infinite loop error from being flagged. However, I think that might actually be the wrong approach: if the inference fails, I think it's still okay to flag the possible infinite loop. This is because it's more likely than not that the person writing the code would expect the loop condition to be true rather than false.

So, please modify the implementation of `infer_condition` so that it also returns `True` when the inference fails.